src/main/java/graphql/language/Field.java

```diff
@@ -50,7 +51,7 @@ protected Field(String name,
                     IgnoredChars ignoredChars,
                     Map<String, String> additionalData) {
         super(sourceLocation, comments, ignoredChars, additionalData);
-        this.name = name;
+        this.name = name == null ? null : Interning.intern(name);
         this.alias = alias;
         this.arguments = ImmutableList.copyOf(arguments);
```

Comment:
the original PR has alias interned but this is a user contributed value - its not a closed set of values